subiquity/client/client.py

```diff
@@ -379,7 +378,9 @@ def header_func():
                 # prompting for confirmation will be confusing.
                 os.system("stty sane")
             journald_listen(
-                [status.event_syslog_id], self.subiquity_event_noninteractive, seek=True
+                [status.event_syslog_id],
+                self.subiquity_event_noninteractive,
+                seek=False,
```

Comment:
Ah yes, thanks for bringing this up. This probably deserves it's own commit to be honest. When `seek=True` the client will jump to the newest event in the journal and only continue to print out journal events from there. The problem is that in the non-interactive case, the server can get pretty far by the time the client connects, and any important warnings or errors sent by the server won't be printed on the console. `seek=False` makes sure that the client will always print every message that the server sent.

Reply:
I think the reason this was like this originally though is that you might in some circumstances get the messages twice. In particular if everything restarts because of a subiquity upgrade. It might only be a problem in dry-run mode though! Can you test this please?

Reply:
I think this is still the case unfortunately. Although my dryrun tests this makes the output repeat 4 times when pesudo updating the snap. Without the change, logging before the connect is definitely missed if we don't update, and displayed only once after the update. Any ideas how to test this in a VM? I was attempting to get the snap to accept the refresh but I realized that will nuke the changes. 

Although, if we output twice - once before and one after the update - isn't that alright? I suppose the output before the refresh wouldn't be interesting most of the time, but it would be if something failed and the refresh didn't happen.

Reply:
> I think this is still the case unfortunately. Although my dryrun tests this makes the output repeat 4 times when pesudo updating the snap. 

4! I wasn't expecting that :-)

> Without the change, logging before the connect is definitely missed if we don't update, and displayed only once after the update. Any ideas how to test this in a VM? I was attempting to get the snap to accept the refresh but I realized that will nuke the changes.

It's hard to test this sort of thing in a VM. you need to push two slightly different versions of the snap to two branches, use livefs-edit to inject the snap from one branch into the iso and then boot with subiquity-channel pointing at the other.

> Although, if we output twice - once before and one after the update - isn't that alright? I suppose the output before the refresh wouldn't be interesting most of the time, but it would be if something failed and the refresh didn't happen.

It's not the worst outcome indeed. Here's something you can try though: change the syslog ids in Server.__init__ to be different if self.updated is set.

Reply:
I ran some tests this morning and when `seek=False` we don't get the first run repeated on refresh. We can see both runs on the serial console, but the screen refreshes on update in the virtual console and we only see the new run. Here's an example run from the serial console: https://paste.ubuntu.com/p/M6gcnPzy27/ . 

>Here's something you can try though: change the syslog ids in Server.init to be different if self.updated is set.

Currently we set the syslogs ids with the process id which means they should get reset on restart always. This seems to be the case poking in the journal. Here's a chunk from the journal output https://paste.ubuntu.com/p/GckFjSNRyv/ (1556 should be the old process and 2038 the new one)

Reply:
> Currently we set the syslogs ids with the process id which means they should get reset on restart always.

Well sure "for real" when updated by snapd this is true but for dry-run updating is simulated by using os.execvp which does not change the pid.